source/main/utils/handleCheckBlockReplayProgress.ts

```diff
@@ -43,35 +44,57 @@ function getProgressType(line: string): GetBlockSyncProgressType | null {
 
 const applicationStartDate = moment.utc();
 
-export const handleCheckBlockReplayProgress = (
-  mainWindow: BrowserWindow,
-  logsDirectoryPath: string
-) => {
-  const filename = 'node.log';
-  const logFilePath = `${logsDirectoryPath}/pub/`;
-  const filePath = path.join(logFilePath, filename);
-  if (!fs.existsSync(filePath)) return;
-
-  const tail = new Tail(filePath);
+const createHandleNewLogLine = (mainWindow: BrowserWindow) => {
+  const lastReportedProgressByType: Record<BlockSyncType, number> = {
+    [BlockSyncType.pushingLedger]: 0,
+    [BlockSyncType.replayedBlock]: 0,
+    [BlockSyncType.validatingChunk]: 0,
+  };
 
-  tail.on('line', (line) => {
+  return (line: string) => {
     if (
       !isItFreshLog(applicationStartDate, line) ||
       !containProgressKeywords(line)
     ) {
       return;
     }
 
-    const percentage = line.match(/Progress:([\s\d.,]+)%/)?.[1];
-    const progressType = getProgressType(line);
-    if (!percentage || !progressType) {
+    const unparsedProgress = line.match(/Progress:([\s\d.,]+)%/)?.[1];
```

Comment:
I've decided to make the variable names more concise as the if in line 70 was getting a bit too big to read.